java-squid/src/main/java/org/sonar/java/se/ConstraintManager.java

```diff
@@ -62,14 +68,33 @@ public SymbolicValue createSymbolicValue(Tree syntaxNode) {
       case INSTANCE_OF:
         result = new SymbolicValue.InstanceOfSymbolicValue(counter);
         break;
+      case MEMBER_SELECT:
+        result = createIdentifierSymbolicValue(((MemberSelectExpressionTree) syntaxNode).identifier());
+        break;
+      case IDENTIFIER:
+        result = createIdentifierSymbolicValue((IdentifierTree) syntaxNode);
+        break;
       default:
-        result = symbolicValueFactory == null ? new SymbolicValue(counter) : symbolicValueFactory.createSymbolicValue(counter, syntaxNode);
-        symbolicValueFactory = null;
+        result = createDefaultSymbolicValue(syntaxNode);
     }
     counter++;
     return result;
   }
 
+  private SymbolicValue createIdentifierSymbolicValue(IdentifierTree identifier) {
+    if (BOOLEAN_CLASS_VALUES.contains(identifier.name()) && identifier.symbolType() != null && identifier.symbolType().is(BOOLEAN_CLASS)) {
```

Comment:
More over if value is TRUE or FALSE it should be assigned not a new symbolic value but (at least in a first approximation IMO) the constant SV true or false already defined. 
This would avoid to create all the mechanics for "getConstraintForPrimitiveType" which adds complexity